src/app/core/services/issue.service.ts

```diff
@@ -154,16 +151,29 @@ export class IssueService {
     );
   }
 
-  private createAndSaveIssueModel(githubIssue: GithubIssue): boolean {
-    const issue = this.createIssueModel(githubIssue);
-    this.updateLocalStore(issue);
-    return true;
+  private createAndSaveIssueModels(githubIssues: GithubIssue[], shouldEmit: boolean = true): Issue[] {
+    const issues: Issue[] = [];
+
+    for (const githubIssue of githubIssues) {
+      const issue = this.createIssueModel(githubIssue);
+      issues.push(issue);
+    }
+    this.updateLocalStore(issues, shouldEmit);
+
+    return issues;
   }
 
-  private deleteIssuesFromLocalStore(ids: number[]): void {
-    ids.forEach((id: number) => {
-      this.getIssue(id).subscribe((issue) => this.deleteFromLocalStore(issue));
-    });
+  private deleteIssuesFromLocalStore(ids: number[], shouldEmit: boolean = true): void {
+    const withoutIssuesToRemove = { ...this.issues };
+    for (const id of ids) {
+      delete withoutIssuesToRemove[id];
+    }
```

Comment:
Good idea to construct the object with all the issues and then remove issues from the object at once, rather than to construct an object with all issues and then remove a single issue for every issue to be removed.